src/app/AttributePathParams.h

```diff
@@ -25,7 +25,7 @@
 
 namespace chip {
 namespace app {
-#if CHIP_CONFIG_ENABLE_READ_CLIENT
+#if defined(CHIP_CONFIG_ENABLE_READ_CLIENT) && CHIP_CONFIG_ENABLE_READ_CLIENT
```

Comment:
If this is warning, that means that the right config headers are not being included, no?  The right fix is to include those headers.